src/System.Private.CoreLib/shared/System/GCMemoryInfo.cs

```diff
@@ -17,7 +17,10 @@ namespace System
         public long MemoryLoadBytes { get; }
 
         /// <summary>
-        /// Total available memory for the GC to use when the last GC ocurred. By default this is the physical memory on the machine, but it may be customized by specifying a HardLimit.
+        /// If the environment variable COMPlus_GCHeapHardLimit is set,
```

Comment:
I would keep the "Total available memory for the GC to use when the last GC occurred." from original the comment. It explains what the API does that is the most important part.

How the API is implemented currently is less interesting. It is very likely to change over time.